### ISSUE TITLE
[Feature request] Extend shape inference tests to cover op version <= 5

### ISSUE THREAD

**Issue by guoyuhong** [onnx/onnx#5289]
### System information

main branch

### What is the problem that this feature solves?

I'm working on Issue https://github.com/onnx/onnx/issues/4160 through PR https://github.com/onnx/onnx/pull/5263 .
Current design works fine for Op version > 5. For Op version <= 5. I found the following error:
![image](https://github.com/onnx/onnx/assets/19584326/43ab2fe7-e86e-4342-a39e-d59d39d536cb)
I looks like the `Reshape` creation arguments do not fit for Op version <= 5.

This issue is used to track this problem in case it is forgotten.

### Alternatives considered

The code change for PR https://github.com/onnx/onnx/pull/5263 could be big, so this issue will be solve later on.

### Are you willing to contribute it (Y/N)

Yes

**Comment by gramalingam** [comment]
I am surprised that this problem does not show up more often. There are other ops where attributes are promoted to become inputs. The test-creation logic, however, has to explicitly indicate which are inputs, which are attributes etc. Eg. the Reduce* ops.

**Comment by gramalingam** [comment]
I think, in the end, users creating a test will probably need to specify actual versions, like in the case of `Reshape`. May be with other utilities like `all_versions('Reshape', least_version=5)`

**Comment by guoyuhong** [comment]
@gramalingam The PR https://github.com/onnx/onnx/pull/5263 is trying to solve the problem to increase test coverage. Before the PR, `onnx/test/shape_inference_test.py` only covers the latest version of Ops. When developers introduced incompatible change, developers often deleted the old test code and only kept the latest test code. The PR introduced a scheme that requires test function to cover all versions of Ops. If incompatible changes are introduced, the developers need to add new test code and keep the old test code.

I'm now trying to solve the inference test for OpVersion <= 5. I fixed the `Reshape` creation problem. But I found that when OpVersion=1, the shape inference seems not inference the output shape(showed in following figure). I found that many Ops  had the same problem when OpVersion==1. I don't know the root cause. Maybe some C++ code needs to be changed.
![image](https://github.com/onnx/onnx/assets/19584326/fe6801ec-f73f-4ef9-8d7b-9b11bae0d68f)
